core/development/ui/Events.ts

```diff
@@ -51,7 +51,8 @@ export function initializeEventListeners() {
   HELL_BOARD_BUTTON?.addEventListener('click', handleBoardButtonPress);
   HEAVEN_BOARD_BUTTON?.addEventListener('click', handleBoardButtonPress);
 
-  SHOP_UPGRADE_SWAPPER?.addEventListener('click', swapShopAndUpgrade);
+  // SHOP_UPGRADE_SWAPPER?.addEventListener('click', swapShopAndUpgrade);
+  SHOP_UPGRADE_SWAPPER?.addEventListener('change', () => swapShopAndUpgrade());
```

Comment:
Why not ` SHOP_UPGRADE_SWAPPER?.addEventListener('change', swapShopAndUpgrade);`?
Also, please remove the commented code.